app/controllers/org_admin/templates_controller.rb

```diff
@@ -68,7 +68,17 @@ def edit
 
       @current = Template.current(@template.dmptemplate_id)
 
-      unless @template == @current
+      if @template == @current 
+        if @template.published?
+          new_version = @template.get_new_version
+          if !new_version.nil?
```

Comment:
just checked the method get_new_version... In an event of failing an exception will be raised (e.g. RecordInvalid, RecordNotSaved) which is not caught here and therefore this controller action will fail badly. Can you add an error handler here or remove save! in favour of save at get_new_version model, please?